numpy/lib/tests/test_financial.py

```diff
@@ -24,6 +24,13 @@ def test_rate_decimal(self):
         rate = np.rate(Decimal('10'), Decimal('0'), Decimal('-3500'), Decimal('10000'))
         assert_equal(Decimal('0.1106908537142689284704528100'), rate)
 
+    def test_rate_returns_nan_when_infeasible_with_decimal(self):
+        # It is impossible to pay off the existing ammount by making further
+        # withdrawl, therefore no possible ``rate`` exists.
+        # see: https://github.com/numpy/numpy/issues/14638
+        rate = np.rate(Decimal(12), Decimal(400), Decimal(10000), Decimal(0))
+        assert_(isinstance(rate, Decimal) and rate.is_nan())
```

Comment:
This is a workaround as ``assert_equal(Decimal('nan'), Decimal('nan'))`` is strictly false. If anyone knows a better way, let me know and I'll add it.